enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,462 @@
+	// After this time, the scores are considered to be invalid by placement. nil means never expire.
+	// The controller owning this resource should keep the scores up-to-date.
+	// +optional
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	ValidUntil *metav1.Time `json:"validUntil" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+}
+
+//AddOnPlacementScoreItem represents the score name and value.
+type AddOnPlacementScoreItem struct {
+	// Name is the name of the score
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// Value is the value of the score. The score range is from -100 to 100.
+	// +kubebuilder:validation:Minimum:=-100
+	// +kubebuilder:validation:Maximum:=100
+	// +required
+	Value int32 `json:"value,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AddOnPlacementScoreList is a collection of AddOnPlacementScore.
+type AddOnPlacementScoreList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is a list of AddOnPlacementScore
+	Items []AddOnPlacementScore `json:"items"`
+}
+```
+### Placement API
+Replace `Name` with `PrioritizerScoreCoordinate` inside `PrioritizerConfig`, to support specifying the customized score to sort clusters.
+
+```golang
+type Placement struct {
+  ...
+	Spec PlacementSpec `json:"spec"`
+}
+
+type PlacementSpec struct {
+  ...
+
+	// PrioritizerPolicy defines the policy of the prioritizers.
+	// If this field is unset, then default prioritizer mode and configurations are used.
+	// Referring to PrioritizerPolicy to see more description about Mode and Configurations.
+	// +optional
+	PrioritizerPolicy PrioritizerPolicy `json:"prioritizerPolicy"`
+}
+
+// PrioritizerPolicy represents the policy of prioritizer
+type PrioritizerPolicy struct {
+  ...
+
+	// +optional
+	Configurations []PrioritizerConfig `json:"configurations,omitempty"`
+}
+
+// PrioritizerConfig represents the configuration of prioritizer
+type PrioritizerConfig struct {
+	// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source.
+	// +kubebuilder:validation:Required
+	// +required
+	PrioritizerScoreCoordinate PrioritizerScoreCoordinate `json:"scoreCoordinate"`
+
+	// Weight defines the weight of the prioritizer score. The value must be ranged in [0,10].
+	// Each prioritizer will calculate an integer score of a cluster in the range of [-100, 100].
+	// The final score of a cluster will be sum(weight * prioritizer_score).
+	// A higher weight indicates that the prioritizer weights more in the cluster selection,
+	// while 0 weight indicates that the prioritizer is disabled.
+	// +kubebuilder:validation:Minimum:=-10
```

Comment:
a minus weight is confusing, i suppose the weight coefficient shoudl be [0, 1]?

Reply:
Want to use a minus weight to indicate "not suggest to select', or "reverse selection". For example, a prioritizer "MostAllocatableMemory" sorts clusters by allocatable memory from high to low, if user set weight to a negative number, he can select the cluster with least allocatable memory.

Reply:
The change to allow negative weighting is interesting.  Can you provide examples where being able to select both for and against an addonplacementscore is valuable?

Reply:
Using https://github.com/open-cluster-management-io/enhancements/pull/33#discussion_r752325410 as an example, user owns 5 clusters, he wants to send the workload to the top one, and also wants to get the last one to check if there is something wrong with that cluster or do something else. With negative weight, he can achieve it by creating 2 placement, one with weight 1 to select the top one, and one with weight -1 to select the last one.

Reply:
> Using #33 (comment) as an example, user owns 5 clusters, he wants to send the workload to the top one, and also wants to get the last one to check if there is something wrong with that cluster or do something else. With negative weight, he can achieve it by creating 2 placement, one with weight 1 to select the top one, and one with weight -1 to select the last one.

can you include this in the doc itself for future us?

Reply:
@deads2k yes, add the description about negative weight in the proposal.